wgpu-hal/src/metal/surface.rs

```diff
@@ -268,9 +268,14 @@ impl crate::Surface<super::Api> for super::Surface {
     ) -> Result<Option<crate::AcquiredSurfaceTexture<super::Api>>, crate::SurfaceError> {
         let render_layer = self.render_layer.lock();
         let (drawable, texture) = autoreleasepool(|| {
-            let drawable = render_layer.next_drawable().unwrap();
```

Comment:
ok, I'm confused. How does this work?
This unwrap didn't panic, so what difference does it make to check if it's `Some` now?

Reply:
That unwrap did indeed panic?

Reply:
Hmm. You are right. I thought we are only exhausting memory in this state.

Reply:
That happens because `MTLDrawable`'s are limited to 3 active at maximum as per Metal documentation. The job that cleans these is the same one that cleans other resources.